lib/parsers/acorn-extensions.js

```diff
@@ -0,0 +1,191 @@
+      this.parseExportSpecifiersMaybe(node);
+    }
+    this.parseExportFromWithCheck(node, exports, true);
+  } else if (this.eat(tt._default)) {
+    // export default ...
+    this.checkExport(exports, "default", this.lastTokStart);
+    let isAsync;
+    if (this.type === tt._function || (isAsync = this.isAsyncFunction())) {
+      let fNode = this.startNode();
+      this.next();
+      if (isAsync) this.next();
+      node.declaration = this.parseFunction(fNode, "nullableID", false, isAsync);
+    } else if (this.type === tt._class) {
+      let cNode = this.startNode();
+      node.declaration = this.parseClass(cNode, "nullableID");
+    } else {
+      node.declaration = this.parseMaybeAssign();
+      this.semicolon();
+    }
+    return this.finishNode(node, "ExportDefaultDeclaration");
+  } else if (this.shouldParseExportDeclaration()) {
+    // export var|const|let|function|class ...
+    node.declaration = this.parseStatement(true);
+    if (node.declaration.type === "VariableDeclaration") {
+      this.checkVariableExport(exports, node.declaration.declarations);
+    } else {
+      this.checkExport(
+        exports,
+        node.declaration.id.name,
+        node.declaration.id.start
+      );
+    }
+    node.specifiers = [];
+    node.source = null;
+  } else {
+    // export { x, y as z } [from '...']
+    node.declaration = null;
+    node.specifiers = this.parseExportSpecifiers(exports);
+    this.parseExportFrom(node, false);
+  }
+  return this.finishNode(node, "ExportNamedDeclaration");
+}
+
+function lookAhead(n) {
+  const old = Object.assign(Object.create(null), this);
+  while (n-- > 0) this.nextToken();
+  const copy = Object.assign(Object.create(null), this);
+  Object.assign(this, old);
+  return copy;
+}
+
+function isExportDefaultSpecifier() {
+  if (this.type !== tt.name) {
+    return false;
+  }
+
+  const lookAhead = this.lookAhead(1);
+  return lookAhead.type === tt.comma ||
+    (lookAhead.type === tt.name &&
+     lookAhead.value === "from");
+}
+
+function parseExportSpecifiersMaybe(node) {
+  if (this.eat(tt.comma)) {
+    node.specifiers.push.apply(
+      node.specifiers,
+      this.parseExportSpecifiers()
+    );
+  }
+}
+
+function parseExportFromWithCheck(node, exports, expect) {
+  this.parseExportFrom(node, expect);
+
+  if (node.specifiers) {
+    for (let i = 0; i < node.specifiers.length; i++) {
+      const s = node.specifiers[i];
+      const exported = s.exported;
+      this.checkExport(exports, exported.name, exported.start);
+    }
+  }
+}
+
+function parseExportFrom(node, expect) {
+  if (this.eatContextual("from")) {
+    node.source = this.type === tt.string
+      ? this.parseExprAtom()
+      : this.unexpected();
+  } else {
+    if (node.specifiers) {
+      // check for keywords used as local names
+      for (let i = 0; i < node.specifiers.length; i++) {
+        const local = node.specifiers[i].local;
+        if (local && (this.keywords.test(local.name) ||
```

Comment:
what's the possible values of `local`? If it's either an object or `undefined` we should be explicit about the object check with `typeof local === "object" && local !== null` as v8 (and I guess others) favor that over a bool check.